src/cve/pipeline/pipeline.py

```diff
@@ -128,14 +122,34 @@ def convert_input_to_df(message: AgentMorpheusEngineInput) -> ControlMessage:
 
         has_sufficient_intel_flags = full_df["has_sufficient_intel_for_agent"]
 
+        intel_score_flag = full_df["intel_score"]
+
         full_df["vulnerable_dependencies"] = vulnerable_dependencies
 
+        if os.environ.get("EXTENDED_VERBOSE_DEBUG", False):
+            logger.debug(f"full_df: %s\n", str(full_df))
+
         # Filter full_df for conditions that should skip the agent
         filtered_df = full_df[[
-            len(vuln_deps) > 0 or (not has_vuln_package_info and has_sufficient_intel)  for vuln_deps,
-            has_vuln_package_info, has_sufficient_intel in zip(vulnerable_dependencies, has_vuln_package_info_flags, has_sufficient_intel_flags)
+            __keep_vuln(
+                run_config.general.generate_intel_score,
+                run_config.general.insist_analysis,
+                run_config.general.intel_low_score,
+                vuln_deps,
+                has_vuln_package_info,
+                has_sufficient_intel,
+                int(intel_score)
+            )
```

Comment:
@tmihalac Please wrap call of `__keep_vuln`  function with bool(),  in order to overcome crash when `generate_intel_score=false`

```suggestion
            bool(__keep_vuln(
                run_config.general.generate_intel_score,
                run_config.general.insist_analysis,
                run_config.general.intel_low_score,
                vuln_deps,
                has_vuln_package_info,
                has_sufficient_intel,
                int(intel_score)
            ))
```

Reply:
Done